src/NUnitFramework/framework/Constraints/Comparers/StreamsComparer.cs

```diff
@@ -39,8 +42,15 @@ public static EqualMethodResult Equal(object x, object y, ref Tolerance toleranc
                     return EqualMethodResult.ComparedEqual;
             }
 
+#if NETFRAMEWORK
             byte[] bufferExpected = new byte[BUFFER_SIZE];
             byte[] bufferActual = new byte[BUFFER_SIZE];
+#else
+            ArrayPool<byte> arrayPool = ArrayPool<byte>.Shared;
+
+            byte[] bufferExpected = arrayPool.Rent(BUFFER_SIZE);
+            byte[] bufferActual = arrayPool.Rent(BUFFER_SIZE);
```

Comment:
request:

Can we move this into the try/finally to help ensure we always return the array backto the pool even in the event an exception is thrown? This can help prevent memory leaks. It might look something like this in practice:

```csharp
byte[] bufferExpected = null;

try {
  bufferExpected = ArrayPool<byte>.Shared.Rent(BUFFER_SIZE);
  
  // other things
} finally {
  if (bufferExpected is not null) {
    ArrayPool<byte>.Shared.Return(bufferExpected, clearArray: true);
  }
}

Reply:
Done :)